emergence_lib/src/graphics/ui/hover_panel.rs

```diff
@@ -0,0 +1,180 @@
+                },
+                OrganismText,
+            ));
+
+            // Crafting stuff
+            parent.spawn((
+                TextBundle {
+                    text: Text::from_sections([
+                        TextSection::new("Inputs: ", key_text_style.clone()),
+                        TextSection::from_style(value_text_style.clone()),
+                        TextSection::new("\nOutputs: ", key_text_style.clone()),
+                        TextSection::from_style(value_text_style.clone()),
+                        TextSection::new("\nActive recipe: ", key_text_style.clone()),
+                        TextSection::from_style(value_text_style.clone()),
+                        TextSection::new("\nStatus: ", key_text_style.clone()),
+                        TextSection::from_style(value_text_style.clone()),
+                    ]),
+                    visibility: Visibility::INVISIBLE,
+                    ..default()
+                },
+                CraftingText,
+            ));
+        })
+        .id();
+
+    commands.entity(right_panel).add_child(hover_panel);
+}
+
+/// Update the information displayed in the hover panel.
+pub fn update_hover_panel(
+    cursor_tile_pos: Res<CursorTilePos>,
+    hover_details: Res<HoverDetails>,
+    mut panel_query: Query<&mut Visibility, With<HoverPanel>>,
+    mut position_query: Query<&mut Text, With<PositionText>>,
+    mut organism_query: Query<
+        (&mut Text, &mut Visibility),
+        (
+            With<OrganismText>,
+            // Avoid conflicting queries
+            Without<PositionText>,
+            Without<HoverPanel>,
+        ),
+    >,
+    mut crafting_query: Query<
+        (&mut Text, &mut Visibility),
+        (
+            // Avoid conflicting queries
+            With<CraftingText>,
+            Without<PositionText>,
+            Without<HoverPanel>,
+            Without<OrganismText>,
+        ),
+    >,
+) {
+    if let Some(cursor_tile_pos) = cursor_tile_pos.0 {
+        // Update visibility of the whole panel
+        *panel_query.single_mut() = Visibility::VISIBLE;
+
+        // Update position text
+        position_query.single_mut().sections[1].value =
+            format!("{}, {}", cursor_tile_pos.x, cursor_tile_pos.y);
+
+        // Update organism text
+        if let Some(organism_details) = &hover_details.0 {
+            let (mut text, mut visibility) = organism_query.single_mut();
+
+            *visibility = Visibility::VISIBLE;
+            text.sections[1].value = format!("{}", organism_details.organism_type);
+
+            // Update crafting text
+            if let Some(crafting_details) = &organism_details.crafting_details {
+                let (mut text, mut visibility) = crafting_query.single_mut();
+                *visibility = Visibility::VISIBLE;
+
+                // Update all text entries for crafting
+                text.sections[1].value = format!("{}", crafting_details.input_inventory);
```

Comment:
You can move the variable inside of the quotes these days :)

Reply:
Unfortunately, this only works if you use the variable directly, if you do the dot access you need to do positional formatting arguments :(